src/components/SelectableAccountsList.js

```diff
@@ -88,20 +114,107 @@ class SelectableAccountsList extends Component<ListProps> {
   }
 }
 
-class SelectableAccount extends PureComponent<{
-  account: Account,
-  onPress?: Account => void,
-  isDisabled?: boolean,
-  isSelected?: boolean,
-}> {
+class SelectableAccount extends PureComponent<
+  {
+    account: Account,
+    onPress?: Account => void,
+    isDisabled?: boolean,
+    isSelected?: boolean,
+    showHint: boolean,
+    rowIndex: number,
+    listIndex: number,
+    navigation: NavigationScreenProp<*>,
+    onAccountNameChange: (name: string, changedAccount: Account) => void,
+  },
+  { stopAnimation: boolean },
+> {
+  state = {
+    stopAnimation: false,
+  };
   onPress = () => {
     const { onPress, account, isSelected } = this.props;
     track(isSelected ? "UnselectAccount" : "SelectAccount");
     if (onPress) onPress(account);
   };
 
+  panResponder: PanResponder;
+  swipeableRow: Swipeable;
+
+  updateRef = ref => {
+    if (ref) this.swipeableRow = ref;
+  };
+
+  renderLeftActions = (progress, dragX) => {
+    const translateX = dragX.interpolate({
+      inputRange: [0, 1000],
+      outputRange: [-112, 888],
+    });
+
+    return (
+      <Animated.View
+        style={[
+          styles.leftAction,
+          { transform: [{ translateX }] },
+          { marginLeft: 12 },
+        ]}
+      >
+        <Button
+          event="HardResetModalAction"
+          type="primary"
+          title="Edit Name"
+          onPress={this.editAccountName}
+          containerStyle={styles.buttonContainer}
+        />
+      </Animated.View>
+    );
+  };
+
+  editAccountName = () => {
+    const { account, navigation, onAccountNameChange } = this.props;
+    swipedAccountSubject.next({ row: -1, list: -1 });
+    navigation.navigate("EditAccountName", {
+      onAccountNameChange,
+      account,
+    });
+  };
+
+  componentDidMount() {
+    swipedAccountSubject.subscribe(msg => {
+      const { row, list } = msg;
+      this.setState({ stopAnimation: true });
+      if (
+        this.swipeableRow &&
+        (row !== this.props.rowIndex || list !== this.props.listIndex)
+      ) {
+        this.swipeableRow.close();
+      }
+    });
+  }
+
+  constructor(props) {
+    super(props);
+    const { rowIndex, listIndex } = this.props;
+    this.panResponder = PanResponder.create({
+      // Ask to be the responder:
+      onStartShouldSetPanResponder: () => true,
+      onStartShouldSetPanResponderCapture: () => false,
+      onMoveShouldSetPanResponder: () => false,
+      onMoveShouldSetPanResponderCapture: () => false,
+
+      onPanResponderGrant: () => {
+        if (swipedAccountSubject) {
```

Comment:
Why the if?

Reply:
I was getting undefined if it triggered while we are still scanning, probably something like the component is there but the ref is not ready yet... I don't know. I needed it.

Reply:
Isn't that variable a global immutable subject? So it's always defined

Reply:
don't know what to tell you, if I remove the check it sometimes crashes